File: front_end/src/app/usuario/usuario.ts
import { Moment } from "moment";

export interface IUsuario {
  idCadUsuario?: number; 
	dcrUsuario?: string;
	dcrLogin?: string;
	dcrSenha?: string;
	datCadastro?: Moment;
	datDesativacao?: Moment;
}

export class Usuario implements IUsuario {
    constructor(
      public idCadUsuario?: number,
      public dcrUsuario?: string,
      public dcrLogin?: string,
      public dcrSenha?: string,
      public datCadastro?: Moment,
      public datDesativacao?: Moment
    ) {}
}

File: back_end/src/main/java/github/com/Educord97/entity/Usuario.java
package github.com.Educord97.entity;

import java.time.Instant;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Usuario {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer idCadUsuario; 
	
	@NotNull
	@Size(max = 45)
	@Column(name = "dcr_usuario")
	private String dcrUsuario;
	
	@NotNull
	@Size(max = 45)
	@Column(name = "dcr_login")
	private String dcrLogin;
	
	@NotNull
	@Size(max = 255)
	@Column(name = "dcr_senha")
	private String dcrSenha;
	
	@NotNull
	@Column(name = "dat_cadastro")
	private Instant datCadastro;
	
	@Column(name = "dat_desativacao")
	private Instant datDesativacao;

}
